### PR TITLE
[Agent] Fix Windows agent handle leaks and optimize the ctrl_ip acquisition logic #21291

### PR DESCRIPTION
**Pull request by TomatoMr**
### This PR is for:

- Agent

#### Steps to reproduce the bug
- 
#### Changes to fix the bug
- Close the handle in time
- Remove redundant logic for obtaining ctrl_ip
#### Affected branches
- main
- v6.1
- v6.2
#### Checklist
- [ ] Added unit test to verify the fix.
- [ ] Verified eBPF program runs successfully on linux 4.14.x.
- [ ] Verified eBPF program runs successfully on linux 4.19.x.
- [ ] Verified eBPF program runs successfully on linux 5.2.x.

### CODE REVIEW

File: agent/src/common/meta_packet.rs [hunk 1]
```diff
@@ -458,6 +458,7 @@ impl<'a> MetaPacket<'a> {
 
         self.header_type = HeaderType::Eth;
         self.vlan_tag_size = vlan_tag_size as u8;
+        self.l2_l3_opt_size = vlan_tag_size as u16;
```
Comment (TomatoMr) [inline agent/src/common/meta_packet.rs:461]: 在 https://github.com/deepflowio/deepflow/blob/main/agent/src/handler/mod.rs#L89 处，有使用 l2_l3_opt_size  计算：
`l3_opt_size: meta_packet.l2_l3_opt_size - meta_packet.vlan_tag_size as u16,`
因此先赋值，避免溢出

File: agent/src/config/handler.rs [hunk 2]
```diff
@@ -762,15 +761,12 @@ impl TryFrom<(Config, RuntimeConfig)> for ModuleConfig {
             // If the environment variable K8S_MEM_LIMIT_FOR_DEEPFLOW is set, its value is preferred as the memory limit
             conf.max_memory = k8s_mem_limit;
         }
-        #[cfg(target_os = "linux")]
```
Comment (TomatoMr) [inline agent/src/config/handler.rs:765]: 在 Windows 环境遇到获取网卡列表失败的情况，去掉每次获取配置都重新获取 ctrl_ip 的逻辑，防止因重复获取 ctrl_ip 而导致 agent 重启的情况

File: agent/src/common/meta_packet.rs [hunk 3]
```diff
@@ -501,7 +502,8 @@ impl<'a> MetaPacket<'a> {
                         .unwrap(),
                 );
                 self.ttl = packet[IPV6_HOP_LIMIT_OFFSET + vlan_tag_size];
-                self.l2_l3_opt_size = vlan_tag_size as u16;
+                // In order not to affect the field offset of layer 4, the 20 more bytes of ipv6 than ipv4 are put into l2_l3_opt_size
+                self.l2_l3_opt_size += IPV6_HEADER_ADJUST as u16;
```
Comment (TomatoMr) [inline agent/src/common/meta_packet.rs:506]: 貌似 agent 这里的实现和 trident 不一样，所以同步了一下

Comment (yuanchaoa) [reply agent/src/common/meta_packet.rs:506]: 这里agent变了 不用修改

Comment (TomatoMr) [reply agent/src/common/meta_packet.rs]: 好的，已经修改回去

File: agent/src/platform/platform_synchronizer/windows.rs [hunk 4]
```diff
@@ -213,7 +227,7 @@ impl PlatformSynchronizer {
     ) -> Result<u64, tonic::Status> {
         let config_guard = process_args.config.load();
         let trident_type = config_guard.trident_type;
-        let ctrl_ip = config_guard.source_ip;
+        let ctrl_ip = process_args.ctrl_ip.read().clone();
```
Comment (rvql) [inline agent/src/platform/platform_synchronizer/windows.rs:230]: 这里为啥要`clone`？

File: agent/src/platform/platform_synchronizer/windows.rs [hunk 5]
```diff
@@ -257,7 +271,7 @@ impl PlatformSynchronizer {
             let config_guard = args.config.load();
             let cur_vtap_id = config_guard.vtap_id;
             let trident_type = config_guard.trident_type;
-            let ctrl_ip = config_guard.source_ip;
+            let ctrl_ip = args.ctrl_ip.read().clone();
```
Comment (rvql) [inline agent/src/platform/platform_synchronizer/windows.rs:274]: 这里也是

File: agent/src/platform/platform_synchronizer/windows.rs (hunk 4 above)
Comment (TomatoMr) [reply agent/src/platform/platform_synchronizer/windows.rs]: 好的，已修改

**Review by rvql — APPROVED** [review]

**Pull request merged**